server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java

```diff
@@ -106,13 +106,15 @@ record SeqNoRange(long fromSeqNo, long toSeqNo) {
 
     public void start() {
         LOGGER.debug("[{}] Spawning the shard changes reader", shardId);
-        newRunnable().run();
+        var retryRunnable = newRunnable();
```

Comment:
minor:given that `RetryRunnable` is both `Runnable` and `Cancellable` now, we can probably drop extra variable `retryRunnable` and assign newRunnable() directly to cancellable?

Reply:
unfortunately not as a `cancellable` does not implement `Runnable`.

Reply:
ah yes, sorry for the confusion